src/systems/triggered_publisher/TriggeredPublisher.hh

```diff
@@ -243,8 +291,136 @@ namespace systems
 
     /// \brief Mutex to synchronize access to publishQueue
     private: std::mutex publishQueueMutex;
+
+    private: class ReplyHandler
+    {
```

Comment:
Given that this whole thing is private, I'd consider putting this into the `.cc` file rather than in the header if at all possible.  That should make it more readable.

Reply:
Yeah these look like implementation details that should be hidden from the user. In general, we use PIMPL style in Gazebo, so things like this should go to a private `dataPtr` variable that points to a private class called `TriggeredPublisherPrivate`. In that class, everything is `public` and accessible to `TriggeredPublisher`. Then you can put these functions in that private class.

You can see examples in other plugins by `grep -rI Private src/systems`, for example this plugin https://github.com/gazebosim/gz-sim/blob/8c9489dd01f44cfe0a66d6aaea458025e87d8c48/src/systems/particle_emitter2/ParticleEmitter2.cc#L45